src/Database/FunctionsBuilder.php

```diff
@@ -158,6 +158,30 @@ public function coalesce($args, $types = [])
         return $this->_build('COALESCE', $args, $types, current($types) ?: 'string');
     }
 
+    /**
+     * Returns a FunctionExpression representing a call to SQL CAST function.
+     *
+     * @param string|\Cake\Database\ExpressionInterface $field Field or expression to cast.
+     * @param string $type The target data type
+     * @return \Cake\Database\Expression\FunctionExpression
+     */
+    public function cast($field, $type = '')
```

Comment:
Why is `$type` defaulted to ''?

Reply:
Because of backwards compatibility. Function is called with only an array in first argument.

Reply:
I think we should check that $type was passed as an argument when $field is not an array.

Reply:
Firstly I think so too, but I didn't add this check because of 2 reason:
1. Both of check or no check will return an exception.
2. The default value for its will be removed when remove deprecation.